httputil/useragent.go

```diff
@@ -60,7 +60,10 @@ func SetUserAgentFromVersion(version string, extraProds ...string) {
 	// xxx this assumes ReleaseInfo's ID and VersionID don't have weird characters
 	// (see rfc 7231 for values of weird)
 	// assumption checks out in practice, q.v. https://github.com/zyga/os-release-zoo
-	userAgent = fmt.Sprintf("snapd/%v (%s)%s %s/%s (%s)", version, strings.Join(extras, "; "), extraProdStr, release.ReleaseInfo.ID, release.ReleaseInfo.VersionID, string(arch.UbuntuArchitecture()))
+	userAgent = fmt.Sprintf("snapd/%v (%s)%s %s/%s (%s %s)", version,
```

Comment:
Yes, I'm quite sure we depend on that string already to produce statistics.

Also, it's quite confusion to have a kernel version completely unadorned on this line. There's no hint that this is talking about the kernel, and this string is getting wild as a whole.